ldap.go

```diff
@@ -0,0 +1,96 @@
+package gimlet
+
+import (
+	"crypto/tls"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ldap "gopkg.in/ldap.v2"
+)
+
+// LDAPAuthenticator provides authentication and authorization of users against an LDAP service.
+type LDAPAuthenticator struct {
+	url  string
+	port string
+	path string
+	conn *ldap.Conn
+}
+
+// NewLDAPAuthenticator constructs an LDAPAuthenticator. It requires a url and port to the LDAP
+// server. It also requires a path to user resources that can be passed to an LDAP query.
+func NewLDAPAuthenticator(url, port, path string) (*LDAPAuthenticator, error) {
```

Comment:
return the interface type, making the authenticator itself not-exported. 

optionally, make the creation options a structure so you don't have to pass 3 strings to a function (easy to get in the wrong order. maybe pass port number as an int to do easier validation?